mixins/test/localize-mixin.test.js

```diff
@@ -297,6 +328,50 @@ describe('LocalizeMixin', () => {
 
 	});
 
+	describe('localizeHTML', async() => {
+
+		const elem = await fixture(`<${localizeHTMLTag}></${localizeHTMLTag}>`);
+
+		it('should replace acceptable markup with correct HTML', async () => {
+			const defaultTags = elem.localizeHTML('test1');
+			const manual = elem.localizeHTML('test2', { link: chunks => markup`<d2l-link href="http://d2l.com">${chunks}</d2l-link>` });
+			const disallowed = elem.localizeHTML('test3', { link: chunks => markup`<div>${chunks}</div>` });
+			const badTemplate = elem.localizeHTML('test3', { link: chunks => html`${chunks}` });
+			const tooltip = elem.localizeHTML('test4', { tooltip: generateTooltipHelp({ contents: 'Tooltip text' }) });
+
+			const items = ['milk'];
+			const pluralLink = elem.localizeHTML('pluralTest', { itemCount: items.length, item: items[0], link: generateLink({ href: 'checkout' }) });
+
+			items.push('bread', 'eggs');
+			const val8 = elem.localizeHTML('pluralTest', { itemCount: items.length, link: generateLink({ href: 'checkout' }), html: () => items.map(i => markup`<p>${i}</p>`)});
+
+			const getRenderString = (data) => {
+				if (data.constructor === String) return data;
+				if (Array.isArray(data)) {
+					return data.map(e => getRenderString(e)).join('');
+				}
+				const { strings, values } = data;
+				const flatValues = [...values, ''].map((v, idx) => {
+					if (typeof v === 'object') return getRenderString(v);
+					if (v.toString() === 'Symbol(lit-nothing)') {
+						return '{lit-nothing}'
+					}
+					return v;
+				});
+				return strings.reduce((acc, s, idx) => acc + s + flatValues[idx], '').replace(/\s[\w-]+="{lit-nothing}"/g, '');
+			}
```

Comment:
This is basically just looping through all the template parts and building up the final string. I did this instead of actually rendering each thing out, as that's not really what we're trying to test here, but I'm not tied to it.